client/components/UI/DateTime.tsx

```diff
@@ -12,37 +12,81 @@ import './style.scss';
  * @name DateTime
  * @description DateTime component to display text formatted date and time
  */
-function DateTime({date, withTime, withDate, withYear, padLeft, toBeConfirmed}) {
+function DateTime({
+    date,
+    withTime,
+    withDate,
+    withYear,
+    padLeft,
+    toBeConfirmed,
+    isFullDay,
+    isEndEventDateTime,
+    noEndTime,
+    multiDay,
+}) {
     const {gettext} = superdeskApi.localization;
     const dateFormat = appConfig.planning.dateformat;
     const timeFormat = appConfig.planning.timeformat;
-    const newTimeFormat = toBeConfirmed ? `[${gettext(`Time ${gettext('TBC')}`)}]` : timeFormat;
-    let format = withYear ? dateFormat : dateFormat.replace(/([\/\#.-]y+)|(y+[\/\#.-])/gi, '');
+    const newTimeFormat = toBeConfirmed
+        ? `[${gettext(`Time ${gettext('TBC')}`)}]`
+        : timeFormat;
+    let format = withYear
+        ? dateFormat
+        : dateFormat.replace(/([\/\#.-]y+)|(y+[\/\#.-])/gi, '');
     let dateTimeFormat = [
         withDate ? format : null,
         withTime ? newTimeFormat : null,
-    ].filter((d) => d).join('\u00a0'); // &nbsp;
+    ]
+        .filter((d) => d)
+        .join('\u00a0'); // &nbsp;
+
+    let eventStartDate;
+
+    if (isFullDay && multiDay) {
+        eventStartDate = moment.utc(date).format(dateFormat);
+    } else if (noEndTime && !multiDay || multiDay) {
+        eventStartDate = moment(date).format(dateFormat) + ' ' + moment(date).format(timeFormat);
```

Comment:
why not `moment(date).format(dateTimeFormat)` ?

Reply:
in some scenarios we are getting only time 09:00 instead of 27.11 09:00 date and time. So, for getting desired result we have to use it

Reply:
what scenarios?

Reply:
When `noEndTime` is `true` or `multiday` is `true` or `false` sometimes we are getting only time through `dateTimeFormat` so to make sure we are getting desired output we are using this

Reply:
can you add an example? I still don't get it

Reply:
![image (2)](https://user-images.githubusercontent.com/73937490/204450590-6efc6e59-e665-437d-8021-bde8750e5d48.png)
![image (3)](https://user-images.githubusercontent.com/73937490/204450598-c692dcda-eace-4635-b1a1-d9ba53266ac1.png)
So, here is an examples. I consoled 
`moment(date).format(dateTimeFormat)`
Case 1:-  `allDay = true` and `multiDay = false`, as you can see even if we are using `dateTime` format we are only getting time "05 : 30".
Case 2:- `noEndTime= true` and `multiDay = false`, we are getting only time not date and time. like:- 18:29, 11:00, 23:59.
We are only getting date and time with `moment(date).format(dateTimeFormat)` when `multiDay` is true.

Reply:
still not sure I get this, my question was why are we using
```
eventStartDate = moment(date).format(dateFormat) + ' ' + moment(date).format(timeFormat);
```
instead of
```
eventStartDate = moment(date).format(dateTimeFormat);
```
?